server/schedule/region_scatterer.go

```diff
@@ -307,15 +324,29 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string) *
 
 	scatterWithSameEngine := func(peers []*metapb.Peer, context engineContext) {
 		stores := r.collectAvailableStores(group, region, context)
+		maxStoreTotalCount := uint64(0)
+		minStoreTotalCount := uint64(math.MaxUint64)
+		for _, store := range r.cluster.GetStores() {
+			count := context.selectedPeer.storeTotalCount(store.GetID())
+			if count > maxStoreTotalCount {
+				maxStoreTotalCount = count
+			}
+			if count < minStoreTotalCount {
+				minStoreTotalCount = count
+			}
+		}
 		for _, peer := range peers {
 			if len(stores) == 0 {
 				context.selectedPeer.reset()
 				stores = r.collectAvailableStores(group, region, context)
 			}
-			if context.selectedPeer.put(peer.GetStoreId(), group) {
-				delete(stores, peer.GetStoreId())
-				targetPeers[peer.GetStoreId()] = peer
-				continue
+			storeCount := context.selectedPeer.storeTotalCount(peer.GetStoreId())
+			if storeCount < maxStoreTotalCount || storeCount == minStoreTotalCount {
```

Comment:
What is the difference between `storeCount < maxStoreTotalCount` and `storeCount == minStoreTotalCount`?

Reply:
If the storeCounts are all the same( maxStoreTotalCount == minStoreTotalCount), we can pick this store as candidate.